backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -1522,6 +1534,7 @@ export class ExperimentAssignmentService {
       }
       await Promise.all(promiseArray);
     } else {
+      const enrollmentCounts = await this.getEnrollmentCountPerCondition(experiment, user.id);
```

Comment:
You should not derive the condition during mark experiment point because the enrollment count will be different during assignment and enrollment. Instead use the condition passed by the client.